lib/BC/PcodeCFG.cpp

```diff
@@ -0,0 +1,166 @@
+                    this->GetBlockExitsForIndex(next_start - 1));
+}
+
+
+namespace {
+struct IntraProcTransferCollector {
+  std::vector<size_t> operator()(const InstrExit &ex) {
+    return std::vector<size_t>();
+  }
+
+  std::vector<size_t> operator()(const IntrainstructionIndex &ex) {
+    return std::vector<size_t>(1, ex.target_block_ind);
+  }
+
+
+  std::vector<size_t> operator()(const Exit &ex) {
+    return std::visit(*this, ex);
+  }
+
+  std::vector<size_t> operator()(const ConditionalExit &ex) {
+    std::vector<size_t> res = std::visit(*this, ex.true_branch);
+    std::vector<size_t> other = std::visit(*this, ex.false_branch);
+    res.insert(res.end(), other.begin(), other.end());
+    return res;
+  }
+};
+}  // namespace
+
+std::vector<size_t> PcodeCFGBuilder::GetIntraProcTargets(size_t index) {
+  std::variant<Exit, ConditionalExit> ex = this->GetBlockExitsForIndex(index);
+  return std::visit(IntraProcTransferCollector{}, ex);
+}
+
+std::variant<Exit, ConditionalExit>
+PcodeCFGBuilder::GetBlockExitsForIndex(size_t index) {
+  CHECK(index < this->linear_ops.size());
+  const auto &curr_op = this->linear_ops[index];
+
+  auto build_direct_target_exit = [](VarnodeData target,
+                                     size_t curr_ind) -> Exit {
+    if (isVarnodeInConstantSpace(target)) {
+      // need to treat as signed?
+      return IntrainstructionIndex{curr_ind + target.offset};
+    } else {
+      return InstrExit{};
+    }
+  };
+  switch (curr_op.op) {
+    case CPUI_BRANCH:
+    case CPUI_CALL: {
+      return Exit{build_direct_target_exit(curr_op.vars[0], index)};
+    }
+    case CPUI_CBRANCH: {
+
+
+      auto taken_exit = build_direct_target_exit(curr_op.vars[0], index);
+
+      Exit fallthrough_exit = InstrExit{};
+      // if we are not the last pcodeop then we have an intraproc fallthrough
+      if (index < this->linear_ops.size() - 1) {
+        fallthrough_exit = IntrainstructionIndex{index + 1};
+      }
+
+      return ConditionalExit{taken_exit, fallthrough_exit};
+    }
+    case CPUI_CALLIND:
+    case CPUI_BRANCHIND: {
+      return Exit{InstrExit{}};
+    }
+
+    default: {
+      return Exit{InstrExit{}};
+    }
+  }
+}
+
+
+PcodeBlock::PcodeBlock(size_t base_index)
+    : base_index(base_index),
+      ops(),
+      block_exit(Exit{InstrExit{}}) {}
+
+PcodeCFG PcodeCFGBuilder::Build() {
+
+  auto starts = this->GetBlockStarts();
+  std::set s(starts.begin(), starts.end());
+  starts.assign(s.begin(), s.end());
+
+  std::unordered_map<size_t, PcodeBlock> blocks;
+
+  if (this->linear_ops.empty()) {
+    // There is no insturction at 0 to build a block at
+    // build an empty block so we transfer through to exit by terminating the block
+    blocks.insert({0, PcodeBlock(0)});
+    return blocks;
+  }
+
+  for (size_t i = 0; i < starts.size(); i++) {
+    auto next_start = this->linear_ops.size();
+    if ((i + 1) < starts.size()) {
+      next_start = starts[i + 1];
+    }
```

Comment:
I would format this as ternary operator, to simplify the flow.

Reply:
[806568c](https://github.com/lifting-bits/remill/pull/645/commits/806568ccd203366a2b7d8e9c23ce06ebeda62467)